src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -68,7 +67,7 @@ public static int IndexOfAny(ref byte searchSpace, int searchSpaceLength, ref by
             return index;
         }
 
-        public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
+        public static unsafe int IndexOf(ref byte searchSpace, ref byte value, int length)
```

Comment:
You should not be passing the value by ref here.

Reply:
I was trying to cast the T to a byte [here](https://github.com/dotnet/corefx/pull/24911/files#diff-3af9c1a70a222e20779798cb10750640R51) and the necessary Unsafe APIs need a ref (`ref Unsafe.As<T, byte>(ref value)`). Is there another way to accomplish this?

Reply:
You can take the ref in the wrapper method if it is the best way to do it. That's fine. But the ref should not leak into the actual non-generic implementation here.

Another way to do it would be to cast T to object, and then cast it to byte, like (byte)(object)value. It does not require unsafe code and it may actually produce a tiny bit better code with the current .NET Core JIT. However, it would not be very good on older runtime that do not have the unboxing optimizations.

Reply:
I don't like underscores, but this would also work `__refvalue(__makeref(value), byte)`